rust/src/storage/azure/mod.rs

```diff
@@ -362,12 +362,21 @@ impl StorageBackend for AdlsGen2Backend {
             .directory(obj.path)
             .into_stream()
             .flat_map(|it| match it {
-                Ok(paths) => Either::Left(stream::iter(paths.into_iter().map(|p| {
-                    Ok(ObjectMeta {
-                        path: format!("adls2://{}/{}", self.file_system_name, path.to_owned()),
-                        modified: p.last_modified,
-                        size: Some(p.content_length),
-                    })
+                Ok(paths) => Either::Left(stream::iter(paths.into_iter().flat_map(|p| {
```

Comment:
Huh I didn't know `flat_map` worked on `Option`. Would have guessed to use `filter_map` usually, and it might be a bit clearer?

```suggestion
                Ok(paths) => Either::Left(stream::iter(paths.into_iter().filter_map(|p| {
```